spacy/cli/debug_data.py

```diff
@@ -46,17 +47,26 @@ def debug_config_cli(
     overrides = parse_config_overrides(ctx.args)
     import_code(code_path)
     with show_validation_error():
+        config = Config().from_disk(config_path)
         nlp, _ = util.load_model_from_config(
-            Config().from_disk(config_path), overrides=overrides, auto_fill=auto_fill
+            config, overrides=overrides, auto_fill=auto_fill
         )
     is_stdout = output_path is not None and str(output_path) == "-"
     msg.good("Config is valid", show=not is_stdout)
+    if auto_fill:
+        orig_config = config.to_str()
+        filled_config = nlp.config.to_str()
+        if orig_config == filled_config:
+            msg.text("Config is up-to-date, no values were auto-filled")
+        else:
+            msg.info("Updated config values")
+        if diff:
+            print(diff_strings(config.to_str(), nlp.config.to_str()))
     if is_stdout:
         print(nlp.config.to_str())
-    elif output_path is not None or auto_fill:
-        filled_output = output_path if output_path is not None else config_path
-        nlp.config.to_disk(filled_output)
-        msg.good("Saved updated config", filled_output)
+    elif output_path is not None:
+        nlp.config.to_disk(output_path)
+        msg.good("Saved updated config", output_path)
 
```

Comment:
I was going through your commits sequentially and I agree that we probably should avoid overwriting stuff (was going to make a comment otherwise). Better to be on the safe side here ;-)

Reply:
Yeah, I got bitten by that immediately, like 10 minutes after writing the code to do it, so I was like... nope, that needs to go again 😅